### PR TITLE
Ensure that specified encoding is used everywhere

### PR DESCRIPTION
**Pull request by DGuidi**
I must admit that I'm not sure about what I'm doing (!!!) but looks that cpgStreamProvider shall be initialized using encoding.WebPage (instead that encoding.EncodingName) so it's possible to rebuilt inside the dbf provider the actual encoding specified in the ShapefileDataReader constructor.

Looks a single line but actually a deep change, so please let me know how do you think.

Related issues #39  and #52

### CODE REVIEW

File: src/NetTopologySuite.IO.ShapeFile/Dbase/DbaseFileHeader.cs [hunk 1]
```diff
@@ -335,7 +335,8 @@ public void ReadHeader(BinaryReader reader, IStreamProvider cpgStreamProvider)
 
             // Last byte is a marker for the end of the field definitions.
             // Trond Benum: This fails for some presumeably valid test shapefiles, so I have commented it out.
-            /*byte lastByte = */reader.ReadByte();//s(1)[0];
```
Comment (DGuidi) [inline src/NetTopologySuite.IO.ShapeFile/Dbase/DbaseFileHeader.cs]: this can be reverted, sorry

**Review by airbreather — APPROVED** [review]
LGTM

**Comment by qiangwai** [comment]
Some shape files do not have .cpg , caused the default encoding to be used when reading?

**Comment by DGuidi** [comment]
you can specify the encoding to be used in the ShapefileDataReader .ctor

**Pull request merged**